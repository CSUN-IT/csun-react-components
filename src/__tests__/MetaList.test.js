import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import shortid from 'shortid';
import MetaList from '../../src/components/meta/List';

configure({ adapter: new Adapter() });

describe('MetaList', () => {
  it('Should be defined', () => {
    expect(MetaList).toBeDefined();
  });

  it('Renders component correctly', () => {
    const listItems = [
      {
        text: 'Side Nav Link',
        to: '#',
      },
      {
        text: 'Side Nav Link',
        to: '#',
      },
      {
        text: 'Side Nav Link',
        to: '#',
      },
    ];

    shortid.generate = jest.fn(() => 0);

    const tree = shallow(
      <MetaList className={'test'} listItems={listItems} modifiers={['arrows', 'hover']} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
