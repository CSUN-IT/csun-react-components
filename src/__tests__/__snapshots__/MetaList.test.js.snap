// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaList Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaList
    className="test"
    listItems={
      Array [
        Object {
          "text": "Side Nav Link",
          "to": "#",
        },
        Object {
          "text": "Side Nav Link",
          "to": "#",
        },
        Object {
          "text": "Side Nav Link",
          "to": "#",
        },
      ]
    }
    modifiers={
      Array [
        "arrows",
        "hover",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="list__item"
        >
          <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>
        </li>,
        <li
          className="list__item"
        >
          <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>
        </li>,
        <li
          className="list__item"
        >
          <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>
        </li>,
      ],
      "className": "test list list--arrows list--hover",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>,
          "className": "list__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Side Nav Link",
            "replace": false,
            "to": "#",
          },
          "ref": null,
          "rendered": "Side Nav Link",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>,
          "className": "list__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Side Nav Link",
            "replace": false,
            "to": "#",
          },
          "ref": null,
          "rendered": "Side Nav Link",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="#"
          >
            Side Nav Link
          </Link>,
          "className": "list__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Side Nav Link",
            "replace": false,
            "to": "#",
          },
          "ref": null,
          "rendered": "Side Nav Link",
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="list__item"
          >
            <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>
          </li>,
          <li
            className="list__item"
          >
            <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>
          </li>,
          <li
            className="list__item"
          >
            <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>
          </li>,
        ],
        "className": "test list list--arrows list--hover",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>,
            "className": "list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Side Nav Link",
              "replace": false,
              "to": "#",
            },
            "ref": null,
            "rendered": "Side Nav Link",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>,
            "className": "list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Side Nav Link",
              "replace": false,
              "to": "#",
            },
            "ref": null,
            "rendered": "Side Nav Link",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="#"
            >
              Side Nav Link
            </Link>,
            "className": "list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Side Nav Link",
              "replace": false,
              "to": "#",
            },
            "ref": null,
            "rendered": "Side Nav Link",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
